academiverse-api/src/main/java/com/academiverse/academiverse_api/service/EnrolmentService.java

```diff
@@ -23,12 +24,46 @@ public class EnrolmentService {
     private final InstructRepository instructRepository;
     private final UserRepository userRepository;
 
-    public BaseResponse<List<Long>> getEligibleStudents(Long instructId) {
+    public BaseResponse<List<EnrolEligibleResponse>> getEligibleStudents(Long instructId) {
         Optional<Instruct> instruct = instructRepository.findById(instructId);
-        BaseResponse<List<Long>> response = new BaseResponse<>();
+        BaseResponse<List<EnrolEligibleResponse>> response = new BaseResponse<>();
+
 
         if (instruct.isPresent()) {
-            response.data = null;
+            Long departmentId = instruct.get().getCourse().getDepartment().getDepartmentId();
+
+            // Get all users
+            List<User> allUsers = userRepository.findAll(); // Fetch all users
+
+            // Get a list of enrolled students
+            List<Long> enrolledStudents = enrolmentRepository.findByInstructInstructId(instruct.get().getInstructId())
+                    .stream()
+                    .map(enrolment -> {
+                        return enrolment.getUser().getUserId();
+                    })
+                    .toList();
+
+            // Filter eligible students based on department and enrollment status
+            List<EnrolEligibleResponse> eligibleResponses = allUsers.stream()
+                    .filter(user -> user.getDepartment().getDepartmentId().equals(departmentId) || enrolledStudents.contains(user.getUserId()))
```

Comment:
Looks good! But I think "**|| enrolledStudents.contains(user.getUserId())**" this condition will never be evaluated because the preceding condition( i.e **user.getDepartment().getDepartmentId().equals(departmentId)** ) will be always true.